app/components/common/navbar/index.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState } from 'react';
+import * as Icons from '@heroicons/react/24/outline';
+
+interface DynamicHeroIconType {
+  name: string;
+  className: string;
+}
+interface NavItemType {
+  icon: string;
+  text: string;
+  href: string;
+  id: number;
+  visibleOnDesktop: boolean;
+}
+
+const DynamicHeroIcon = ({ name, className }: DynamicHeroIconType) => {
+  const IconComponent = Icons[`${name}Icon` as keyof typeof Icons];
+
+  if (!IconComponent) {
+    // Return a default one
+    return <Icons.ArrowPathIcon />;
+  }
+
+  return <IconComponent className={className} />;
+};
+
+const Navbar = () => {
+  const [active, setActive] = useState<number>(0);
+
+  const myColor = (elementPosition: number) => {
+    if (active === elementPosition) return true;
+    return false;
+  };
+  const toggle = (elementPosition: number) => {
+    if (active !== elementPosition) setActive(elementPosition);
+  };
+
+  const navbarElements = {
+    arr1: [
+      {
+        icon: 'Home',
+        text: 'Home',
+        href: '/',
+        id: 0,
+        visibleOnDesktop: true,
+      },
+      {
+        icon: 'Calendar',
+        text: 'Events',
+        href: '/',
+        id: 1,
+        visibleOnDesktop: true,
+      },
+      {
+        icon: 'Users',
+        text: 'Social Events',
+        href: '/',
+        id: 2,
+        visibleOnDesktop: true,
+      },
+      {
+        icon: 'Cog6Tooth',
+        text: 'Settings',
+        href: '/',
+        id: 3,
+        visibleOnDesktop: false,
+      },
+    ],
+    arr2: [
```

Comment:
What does arr1,arr2 here means, could we name a better key name here

Reply:
yes, i will change those